pkg/executor/aggfuncs/builder.go

```diff
@@ -144,6 +144,25 @@ func buildApproxCountDistinct(aggFuncDesc *aggregation.AggFuncDesc, ordinal int)
 	return nil
 }
 
+func getEvalTypeForApproxPercentile(aggFuncDesc *aggregation.AggFuncDesc) types.EvalType {
+	arg0FieldType := aggFuncDesc.Args[0].GetType()
+	hasEnumSetAsIntFlag := (arg0FieldType.GetFlag() & mysql.EnumSetAsIntFlag) > 0
+	if hasEnumSetAsIntFlag {
+		arg0FieldType.DelFlag(mysql.EnumSetAsIntFlag)
+	}
+
+	evalType := arg0FieldType.EvalType()
+	if aggFuncDesc.Args[0].GetType().GetType() == mysql.TypeBit {
+		evalType = types.ETString // same as other aggregate function
+	}
+
+	if hasEnumSetAsIntFlag {
+		arg0FieldType.SetFlag(mysql.EnumSetAsIntFlag)
+	}
+
+	return evalType
+}
```

Comment:
Can you add some comments about why deleting `mysql.EnumSetAsIntFlag` and setting it later?

Reply:
> Can you add some comments about why deleting `mysql.EnumSetAsIntFlag` and setting it later?

Because maybe this flag will be used in other place. I will add comment.

Reply:
Why does the original logic cause tidb to crash?